haskell/src/Monocle/Client.hs

```diff
@@ -90,3 +96,22 @@ monocleReq path MonocleClient {..} body =
     decodeResponse body' = case JSONPB.eitherDecode body' of
       Left err -> error $ "Decoding of " <> show body <> " failed with: " <> show err
       Right a -> a
+
+-- Retry 5 times network action, doubling backoff each time
+retry :: (MonadMask m, MonadLog m, MonadIO m) => m a -> m a
```

Comment:
The Prelude would have been a good place too. As long as the code doesn't depends on Monocle, i'd be happy to build reusable modules.